src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHouseLookupDirectService.cs

```diff
@@ -20,9 +20,34 @@ public CompaniesHouseLookupDirectService(HttpClient httpClient)
     {
         var response = await _httpClient.GetAsync($"{CompaniesHouseEndpoint}/{id}");
 
-        if (response.StatusCode == HttpStatusCode.NoContent || response.StatusCode == HttpStatusCode.NotFound || response.StatusCode == HttpStatusCode.InternalServerError)
+        if (response.StatusCode == HttpStatusCode.NoContent || response.StatusCode == HttpStatusCode.NotFound)
         {
-            return null;
+            return new Company
+            {
+               Error = new UploadFileErrorModel
+                {
+                   FileLineNumber = 1,
+                   FileContent = string.Empty,
+                   Message = BulkUpdateErrors.ResourceNotFoundErrorMessage,
+                   ErrorNumber = BulkUpdateErrors.ResourceNotFoundError,
+                   IsError = true
+                }
+            };
+        }
+
+        if (response.StatusCode == HttpStatusCode.InternalServerError)
```

Comment:
InternalServerError will be returned if the companies house number is not 8 characters, so we need to treat it as not found on line 23.

Reply:
I have moved the 8 Character logic to field validations as discussed.